core/store/orm/orm.go

```diff
@@ -485,24 +485,58 @@ func (orm *ORM) AnyJobWithType(taskTypeName string) (bool, error) {
 	return found, ignoreRecordNotFound(rval)
 }
 
-// CreateTx saves the properties of an Ethereum transaction to the database.
+// CreateTx returns a transaction by its surrogate key, if it exists, or
+// creates it and its attempts
 func (orm *ORM) CreateTx(
-	from common.Address,
-	nonce uint64,
-	to common.Address,
-	data []byte,
-	value *big.Int,
-	gasLimit uint64,
+	surrogateID null.String,
```

Comment:
Is this always `jobRunID`? If so, I would prefer the more definitive `jobRunID` than the more abstract `surrogateID`. We're effectively establishing a 1-to-1 relationship if I'm not mistaken.

Or perhaps it can be null for testing purposes... Leave it as is then.

Reply:
It's null in a lot of places, unfortunately.